src/Element/Select2.php

```diff
@@ -236,6 +264,18 @@ public static function preRenderAutocomplete($element) {
     return $element;
   }
 
+  /**
+   * Sets the autocomplete route parameters.
+   *
+   * @param array $element
+   *   The render element.
+   */
+  protected static function setAutocompleteRouteParameters(array &$element) {
+    $complete_form = [];
+    $element = EntityAutocomplete::processEntityAutocomplete($element, new FormState(), $complete_form);
+    $element['#autocomplete_route_name'] = 'select2.entity_autocomplete';
+  }
```

Comment:
We should keep the pattern used by "processEntityAutocomplete" and return the $element